src/Vatras/Translation/LanguageMap.lagda.md

```diff
@@ -348,3 +359,21 @@ OC-is-sound {F} _==_ = soundness-by-expressiveness (2CC-is-sound _==_) (OC-to-2C
 FST-is-sound : ∀ {F : 𝔽} (_==_ : DecidableEquality F) → Sound (FSTL F)
 FST-is-sound {F} _==_ = soundness-by-expressiveness VariantList-is-Sound (FST-to-VariantList.VariantList≽FST F _==_)
 ```
+
+Variation Trees assume variants to be forests of rose trees.
+We hence cannot directly integrate it into the circle of compilers above.
+Yet, variant lists and ADTs are generic in their type of variants and hence can also denote forests.
+```agda
+open import Vatras.Lang.VariantList.Properties
+  using ()
+  renaming (VariantList-is-Sound to VariantList-is-sound-on; VariantList-is-Complete to VariantList-is-complete-on)
+
+ADT-is-sound-on : ∀ {F : 𝔽} (V : 𝕍) (_==_ : DecidableEquality F) → Sound-on V (ADTL F V)
+ADT-is-sound-on {F} V _==_ = soundness-by-expressiveness-on V (VariantList-is-sound-on V) (ADT-to-VariantList.VariantList≽ADT F V _==_)
+
+VT-is-complete : ∀ (F : 𝔽) (f : F) → Complete-on Forest (VTL (Indexed F))
```

Comment:
Should we prove something like `Vatras.Translation.Lang.2CC.Rename` for variation trees such that this reduces to `Complete-on Forest (VTL F)`?

Reply:
Yes that would be great. I guess here it is simpler because we just need an embedding of natural numbers into any set of feature names F? We do not have to do this as part of this PR though.

Reply:
I created issue #87 for this.